sqlstore/src/test/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLStudyDAOTest.java

```diff
@@ -0,0 +1,254 @@
+import uk.ac.bbsrc.tgac.miso.core.store.ExperimentStore;
+import uk.ac.bbsrc.tgac.miso.core.store.ProjectStore;
+import uk.ac.bbsrc.tgac.miso.core.store.Store;
+
+public class SQLStudyDAOTest extends AbstractDAOTest {
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Autowired
+  @Spy
+  private JdbcTemplate jdbcTemplate;
+
+  @Mock
+  private Store<SecurityProfile> securityProfileDAO;
+
+  @Mock
+  private ProjectStore projectStore;
+
+  @Mock
+  private SecurityStore securityDAO;
+
+  @Mock
+  private ChangeLogStore changeLogDAO;
+
+  @Mock
+  private ExperimentStore experimentDAO;
+
+  @Mock
+  private MisoNamingScheme<Study> namingScheme;
+
+  @InjectMocks
+  private SQLStudyDAO dao;
+
+  // Auto-increment sequence doesn't roll back with transactions, so must be tracked
+  private static long nextAutoIncrementId = 5L;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    dao.setJdbcTemplate(jdbcTemplate);
+    dao.setDataObjectFactory(new TgacDataObjectFactory());
+  }
+
+  @Test
+  public void testSaveNull() throws IOException {
+    exception.expect(NullPointerException.class);
+    dao.save(null);
+  }
+
+  @Test
+  public void testSaveNew() throws IOException, MisoNamingException {
+    long autoIncrementId = nextAutoIncrementId;
+    Study newStudy = makeStudy();
+    mockAutoIncrement(autoIncrementId);
+    Mockito.when(namingScheme.validateField(Matchers.anyString(), Matchers.anyString())).thenReturn(true);
+
+    assertEquals(autoIncrementId, dao.save(newStudy));
+
+    Study savedRun = dao.get(autoIncrementId);
+    assertEquals(newStudy.getAlias(), savedRun.getAlias());
+    nextAutoIncrementId += 1;
+  }
+
+  @Test
+  public void testListAll() {
+    List<Study> studies = dao.listAll();
+    assertEquals(4, studies.size());
+  }
+
+  @Test
+  public void testListAllWithLimit() throws IOException {
+    List<Study> studies = dao.listAllWithLimit(2L);
+    assertEquals(2, studies.size());
+  }
+
+  @Test
+  public void testListAllWithBiggerLimit() throws IOException {
+    List<Study> studies = dao.listAllWithLimit(50L);
+    assertEquals(4, studies.size());
+  }
+
+  @Test
+  public void testListAllWithZeroLimit() throws IOException {
+    List<Study> studies = dao.listAllWithLimit(0L);
+    assertEquals(0, studies.size());
+  }
+
+  @Test
+  public void testStudyCount() throws IOException {
+    assertEquals(4, dao.count());
+  }
+
+  @Test
+  public void testListBySearchOICR() {
+    List<Study> runs = dao.listBySearch("OICR");
+    assertEquals(2, runs.size());
+  }
+
+  @Test
+  public void testListBySearchStu() {
+    List<Study> runs = dao.listBySearch("STU");
+    assertEquals(4, runs.size());
+  }
+
+  @Test
+  public void testListBySearchNone() {
+    List<Study> runs = dao.listBySearch("pizza");
+    assertEquals(0, runs.size());
+  }
+
+  @Test
+  public void testListBySearchEmpty() {
+    List<Study> studies = dao.listBySearch("");
```

Comment:
The search adds a SQL wildcard %. So searching for something that doesn't exist returns nothing and searching for nothing returns everything.